src/e2e-test/java/io/cdap/plugin/gcscopy/stepsdesign/GCSCopy.java

```diff
@@ -0,0 +1,158 @@
+
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcscopy.stepsdesign;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.StorageException;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.StorageClient;
+import io.cdap.plugin.common.stepsdesign.TestSetupHooks;
+import io.cdap.plugin.gcscopy.actions.GCSCopyActions;
+import io.cucumber.java.en.Then;
+import org.junit.Assert;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+
+/**
+ * GCS Copy plugin related stepsdesign.
+ */
+
+public class GCSCopy {
+
+  @Then("Validate GCSCopy successfully copies object {string} to destination bucket")
+  public void validateGCSCopySuccessfullyCopiedObjectToDestinationBucket(String path) throws IOException {
+    String sourceGCSBucket = TestSetupHooks.gcsSourceBucketName;
+    String gcsObject = PluginPropertyUtils.pluginProp(path);
+    boolean isPresentatsource = false;
+    for (Blob blob : StorageClient.listObjects(sourceGCSBucket).iterateAll()) {
+      if (blob.getName().contains(gcsObject)) {
```

Comment:
Changed to equals check

Reply:
It was not working with equals for the scenario when sub directories toggle button is set to true, so changed to contains check. As it has to check the subdirectories also.

Reply:
I think in case of subdirectories, we need to check if all the files in the subdirectory (in the fixtures) are present in the source bucket. 

In this context, using `String.contains` to check if the path of any of the objects in the bucket contains (as a substring) the path of the directory containing the test CSV files, seems to be fundamentally incorrect.

I think we need to restructure this check to work correctly in case of subdirectories, rather than trying to make this check pass with a substring (`String.contains`) check.

Reply:
Hi, We have created two separate validation steps for subdirectories set to true and set to false.